### DISCUSSION TITLE
CPU acceleration on amd cluster

### DISCUSSION THREAD

**Discussion by csccva** [AdaptiveCpp/AdaptiveCpp#889] [Q&A] [answered]
Hello,

Hopefully I will be able to compile hipsycl the option  `configopts += '-DWITH_CPU_BACKEND=ON ' ` 

Is there any way to check if it really works? 

Cristian

**Comment by fodinabor** [comment, marked as answer]
Hi @csccva,
depending on whether you are asking regarding `-DWITH_ACCELERATED_CPU=ON` or actually generally a CPU backend:
1. CPU backend in general: compile any example code with just `--hipsycl-targets=omp`, so no GPU code is generated. If you can run the example, the CPU backend is working. E.g. `syclcc -o example example.cpp --hipsycl-targets=omp && ./example` 
2. If you care about the compiler accelerated CPU backend, i.e. `-DWITH_ACCELERATED_CPU=ON`, checking whether it's actually used can e.g. be done by setting the environment variable `HIPSYCL_DEBUG_LEVEL=3` while building an application with hipSYCL. It will result in very verbose output when actually using the compiler transformations (note this only happens if you use `parallel_for` kernels using `nd_range`). One thing that indicates the compiler transformations are doing their thing, is if the output contains something like the following (obviously with a different function name): `[SubCFG] Form SubCFGs in _ZN7hipsycl4glue12omp_dispatch20iterate_nd_range_ompILi1EZZ4mainENK3$_0clERNS_4sycl7handlerEEUlNS4_7nd_itemILi1EEEE_JEEEvT0_OKNS4_2idIXT_EEENS4_5rangeIXT_EEESG_SC_mPvRSt8functionIFvvEEDpRT1_`

**Comment by csccva** [comment]
Hello,

Sorry for the confusion I had in mind the ` -DWITH_ACCELERATED_CPU=ON ` option. 
 Thanks.

**Comment by csccva** [comment]
Hello,

I got this:
```
[SubCFG] Form SubCFGs in _ZN7hipsycl4glue12omp_dispatch20iterate_nd_range_ompILi1EZZ4mainENKUlRNS_4sycl7handlerEE_clES5_EUlNS3_7nd_itemILi1EEEE_JEEEvT0_OKNS3_2idIXT_EEENS3_5rangeIXT_EEESG_SC_mPvRSt8functionIFvvEEDpRT1_
```
The output seems the same as yours. It was vector addition code.

It seems that at on LUMI one can install hipSYCL with the rocm/5.1.4.

Cristian
